### ISSUE TITLE
irust doesn't display Thai text properly.

### ISSUE THREAD

**Issue by veer66** [sigmaSd/IRust#127]
![image](https://github.com/sigmaSd/IRust/assets/370688/45f72091-f6b5-45c2-adae-8041255b5f53)

In the left pane I ran irust from git on Konsole. It didn't display upper level Thai characters, which is expected to be displayed like in the right pane.

The string that I use for testing is ผลิตภัณฑ์นี้.

**Comment by sigmaSd** [comment]
Yes seems its a hack I used, because unicode emojis gave me a lot of trouble. with this is at least it, It doesn't mess the cursor mouvement , and other stuff. https://github.com/sigmaSd/IRust/blob/master/crates/printer/src/printer/writer.rs#L76

I'm going to consider this blocked by https://github.com/sigmaSd/IRust/issues/104
